cozy/typecheck.py

```diff
@@ -368,6 +368,12 @@ def visit_EUnaryOp(self, e):
         else:
             raise NotImplementedError(e.op)
 
+    def visit_ESorted(self, e):
+        self.visit(e.e)
+        self.ensure_type(e.asc, BOOL)
+        self.visit(e.asc)
+        e.type = syntax.TList(e.e.type.elem_type)
```

Comment:
You should use `self.get_collection_type` so that
 - this code reports an error if `e.e` was not a collection
 - this code does not crash if `e.e` was not a collection

```suggestion
        e.type = syntax.TList(self.get_collection_type(e.e))
```